config/config_test.go

```diff
@@ -14,6 +14,105 @@ import (
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "/path/v1"},
+		},
+		{
+			"Without http://",
+			ExternalCache{Host: "www.google.com", Path: "/path/v1"},
+			testOut{expectError: false, expectedHost: "www.google.com", expectedPath: "/path/v1"},
+		},
+		{
+			"No scheme but '//' prefix",
+			ExternalCache{Host: "//www.google.com", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "/path/v1"},
+		},
+		{
+			"// appears twice",
+			ExternalCache{Host: "//www.google.com//", Path: "path/v1"},
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "//path/v1"},
+		},
+		{
+			"Host has an only // value",
+			ExternalCache{Host: "//", Path: "path/v1"},
+			testOut{expectError: true, expectedHost: "path", expectedPath: "/v1"},
+		},
+		{
+			"only scheme host, valid path",
+			ExternalCache{Host: "http://", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "", expectedPath: "/path/v1"},
+		},
+		{
+			"No host, path only",
+			ExternalCache{Host: "", Path: "path/v1"},
+			testOut{expectError: true, expectedHost: "", expectedPath: "path/v1"},
+		},
+		{
+			"No host, nor path",
+			ExternalCache{Host: "", Path: ""},
+			testOut{expectError: false, expectedHost: "", expectedPath: ""},
+		},
+		{
+			"Invalid http at the end",
+			ExternalCache{Host: "www.google.com", Path: "http://"},
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "/http://"},
+		},
+		{
+			"Scheme longer than host",
+			ExternalCache{Host: "unknownscheme://host", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "host", expectedPath: "/path/v1"},
+		},
+		{
+			"Wrong colon side in scheme",
+			ExternalCache{Host: "http//:www.appnexus.com", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "", expectedPath: "http//:www.appnexus.com/path/v1"},
+		},
+		{
+			"Missing '/' in scheme",
+			ExternalCache{Host: "http:/www.appnexus.com", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "http:", expectedPath: "http//:www.appnexus.com/path/v1"},
+		},
+		{
+			"host with scheme, no path",
+			ExternalCache{Host: "http://www.appnexus.com", Path: ""},
+			testOut{expectError: true, expectedHost: "www.appnexus.com", expectedPath: ""},
+		},
+		{
+			"scheme, no host nor path",
+			ExternalCache{Host: "http://", Path: ""},
+			testOut{expectError: true, expectedHost: "", expectedPath: ""},
+		},
+	}
+	for _, test := range testCases {
+		var errs configErrors
+		errs = test.data.validate(errs)
+
+		if test.out.expectError {
+			assert.Greater(t, len(errs), 0, "Test case was supposed to throw an error. Desc: %s \n", test.desc)
+			continue
+		} else {
+			assert.Equal(t, 0, len(errs), "Test case was supposed to NOT throw an error. Desc: %s errMsg = %v \n", test.desc, errs)
+		}
+
+		assert.Equal(t, test.out.expectedHost, test.data.Host, "Test case host does not match. Desc: %s  \n", test.desc)
```

Comment:
Consider moving the host and path asserts to the else clause to avoid a continue.